app/lib/core/models/receiver_info.dart

```diff
@@ -22,6 +22,25 @@ class ReceiverInfo {
       @required this.destinationAddress,
       @required this.receiverContactInfo});
 
+  ReceiverInfo.fromJson(Map<String, dynamic> json)
+      : receiverCompanyName = json['receiverCompanyName'],
+        receiverName = json['receiverName'],
+        accountId = json['accountId'] ?? Optional.empty(),
+        destinationLocationId = json['destinationLocationId'],
+        destinationLocationName = json['destinationLocationName'],
+        destinationAddress = Address.fromJSON(json['destinationAddress']),
+        receiverContactInfo = json['receiverContactInfo'];
+
+  Map<String, dynamic> toJSON() => {
+        'receiverCompanyName': receiverCompanyName,
+        'receiverName': receiverName,
+        'accountId': receiverCompanyName ?? Optional.empty(),
```

Comment:
Should not be company name for account ID
```suggestion
        'accountId': accountId ?? Optional.empty(),
```

Reply:
Does the optional translate to JSON? It might need to have toJSON as well...

Reply:
Pretty sure no need to add toJSON here. I tried it and it complained

Reply:
The `Optional` type isn't a JSON primitive, I think we need to do something besides give it optional.

Reply:
My apologies, I forgot the JSON basics. https://stackoverflow.com/questions/9619852/what-is-the-convention-in-json-for-empty-vs-null.

Null is fine, but what we should do here is:
`accountID: acountID.isPresent() ? acccountID.get() : null`